### ISSUE TITLE
to2.4 branch: bomex case crashes, divide-by-0

### ISSUE THREAD

**Issue by fjansson** [dalesteam/dales#32]
Hi All,

I just noticed that the bomex case doesn't work in the to2.4 branch. 
Default options, gnu compiler, debug build. Backtrace:
```
#3  0x82621a in solmpj
	at /home/jansson/code/official-dales/src/modpois.f90:290  <---
#4  0x82905a in __modpois_MOD_poisson
	at /home/jansson/code/official-dales/src/modpois.f90:98
#5  0xb5fdf1 in dales
	at /home/jansson/code/official-dales/src/program.f90:253
#6  0xb5ff1e in main
	at /home/jansson/code/official-dales/src/program.f90:103
```

This is the crashing line, in modpois.f90:290, where z was 0. 
```
p(i,j,kmax) = (p(i,j,kmax)-ak*p(i,j,kmax-1))/z
```
z is calculated a few lines up, from these values:
```
z        = bbk-ak*d(i,j,kmax-1)
bbk = -0.0005447011209443629
ak = 0.0005447011209443629
d() = -1
```

Other cases, like fog, rico, example are not affected. From bisection, it seems the crash starts occurring with the merge 60eceb46eaf7, but I don't know in which commit since I don't manage to compile these versions which use a different CMakeList.

**Comment by fjansson** [comment]
Another crash: thermocrash in Rico case. Introduced in or before e85913ae16.
```
Time of Day: 152252.020    Time of Simulation:     25.60    dt: 12.800000000
Courant numbers (x,y,z,tot):  7.72E-01  1.68E-01  7.60E-01  126  1.05E+00  126
Cell Peclet number:  6.39E-02    2
divmax, divtot =    6.16E-02  -3.25E-10
 thermo crash: i,j,k,niter,thl0(i,j,k),qt0(i,j,k)
          71          52         126           4   459.79906957467978        1.3916960847945342E-003
 thermo crash: i,j,k,niter,thl0(i,j,k),qt0(i,j,k)
At line 515 of file /home/jansson/code/official-dales/src/modthermodynamics.f90
Fortran runtime error: Index '2566' of dimension 1 of array 'ttab' above upper bound of 2000

Error termination. Backtrace:
          36          23         126           4   463.26140361107673        1.3911861253717863E-003
At line 515 of file /home/jansson/code/official-dales/src/modthermodynamics.f90
Fortran runtime error: Index '7623' of dimension 1 of array 'ttab' above upper bound of 2000
```

**Comment by fjansson** [comment]
Commit e3c1ec8d7, the change in modpois.f90 looks suspect.
```
         z        = bbk-ak*d(i,j,kmax-1)
-        if(z/=0.) then
+        if(abs(z)<eps1) then
           p(i,j,kmax) = (p(i,j,kmax)-ak*p(i,j,kmax-1))/z
         else
           p(i,j,kmax) =0.
```

The < should be >, but maybe even better to leave as it was.

**Comment by fjansson** [comment]
Fixed in commit e6e2f27a367b557c6e
